src/main/java/org/apache/maven/plugins/jlink/JLinkMojo.java

```diff
@@ -428,11 +437,12 @@ private JLinkExecutor getExecutor()
         return getJlinkExecutor();
     }
 
-    private boolean projectHasAlreadySetAnArtifact()
+    private boolean projectHasAlreadySetAnArtifact( File createZipArchiveFromImage )
     {
-        if ( getProject().getArtifact().getFile() != null )
+        File projectArtifact = getProject().getArtifact().getFile();
+        if ( projectArtifact != null && projectArtifact.isFile() )
         {
-            return getProject().getArtifact().getFile().isFile();
+            return projectArtifact.getName().equals( createZipArchiveFromImage.getName() );
```

Comment:
@elharo
1. I changed it back to a `@Nullable`  File.
2.  I think I should iterate the Set `getProject().getArtifacts()` here, if I am not mistaken? There can be more than one artifact if I read the methods correctly.